doc/yaml-reference.md

```diff
@@ -90,6 +109,47 @@ aliases:
       - echo $FOO $BAR
 ```
 
+Aliases can extend the top-level `docker_args`. The following example will
+produce the docker arguments `--privileged -v /tmp/bar:/tmp/bar` when executing
+the `example` alias:
+
+```yaml
+docker_args: --privileged
+aliases:
+  example:
+    docker_args: -v /tmp/bar:/tmp/bar
+    script:
+      - ls -l /tmp/
+```
+
+Aliases can also opt to override the top-level `docker_args`, replacing it with
+a new value. This is achieved with the `!override` tag:
+
+```yaml
+docker_args: -v /tmp/foo:/tmp/foo
+aliases:
+  example:
+    docker_args: !override -v /tmp/bar:/tmp/bar
+    script:
+      - ls -l /tmp/
+```
+
+The content of the `docker_args` key is re-parsed as YAML in order to allow
+combining the `!override` tag with other tags; however, this requires quoting
+the value, since YAML forbids a plain-style scalar from beginning with a `!`
+(see [the spec](https://yaml.org/spec/1.2/spec.html#id2788859)). In the next
+example, the top-level alias is replaced with an explicit `!!null` tag, so
+that no additional arguments are passed to docker when executing the `example`
+alias:
+
+```yaml
+docker_args: -v /tmp/foo:/tmp/foo
+aliases:
+  example:
+    docker_args: !override '!!null'
```

Comment:
You have so many tests, I hate to ask: Can we add a test for this?

Reply:
I had one at first, but I ended up removing it, since it's functionally the same as `test_alias_docker_args_override_from_yaml` - they both would be nesting another tag inside of an `!override` node. 

I'll put it in if you insist, but it seems like a "duplicate" to me.

Reply:
I agree that it's effectively the same test. My motivation for adding such a test are:
- I wasn't familiar with `!!null` so ensuring that it works as expected (and doesn't require e.g. opting into)
- Ensuring that the documented examples work as advertised

You've done more than enough, so don't feel obligated; I can address this later.

Reply:
Actually, can you help me understand why using `!!null`:
```yaml
aliases:
  foo:
    docker_args: !override '!!null'
```

...is better than a simple empty string?
```yaml
aliases:
  foo:
    docker_args: !override
```

Reply:
Not necessarily better, but it's an instance of "explicit over implicit". It seems to me the `!!null` would be more obvious to a reader unfamiliar with the nuances in play as to what the intended result is.